File: 2021/09/18/max_product.py
"""
https://leetcode.com/problems/maximum-product-of-three-numbers/
Author: Andrew Jarombek
Date: 9/18/2021
"""

from typing import List


def maximum_product(nums: List[int]) -> int:
    nums.sort()
    return max(nums[0] * nums[1] * nums[-1], nums[-1] * nums[-2] * nums[-3])

File: 2022/04/03/maxProduct.ts
/**
 * https://leetcode.com/problems/maximum-product-of-three-numbers/
 * @author Andrew Jarombek
 * @since 4/3/2022
 */

function maximumProduct(nums: number[]): number {
    nums.sort((a, b) => a - b);

    let option1 = nums[0] * nums[1] * nums[nums.length - 1];
    let option2 = nums[nums.length - 3] * nums[nums.length - 2] * nums[nums.length - 1];
    return Math.max(option1, option2);
}